geode-redis/src/main/java/org/apache/geode/redis/internal/Subscriber.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.apache.geode.redis.internal;
+
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+
+class Subscriber {
+  public final Client client;
+  public final String channel;
+  private ExecutionHandlerContext context;
+
+  public Subscriber(Client client, String channel,
+      ExecutionHandlerContext context) {
+
+    this.client = client;
+    this.channel = channel;
+    this.context = context;
+  }
+
+  public boolean isEqualTo(String channel, Client client) {
+    return channel.equals(this.channel) && client.equals(this.client);
```

Comment:
There is no null checking happening. Can we really assume that the input `channel` and `client` are non-null?

Reply:
Null checking is handled by the respective `equals` methods.

Reply:
So, you can 100% guarantee that the fields are ALWAYS non-null?

Reply:
another approach is that the equals method is `this.channel.equals(channel`. This way the constructor can ensure that the instance channel field is non-null AND your `isEqualsTo` is null-safe

Reply:
Yes, you're correct.

Reply:
This implementation is not "optimal". If you makes sure the `channel` and `client` on the `Subscriber` is non-null, then your `isEqualsTo` method implementation changes. Then the comparison `this.channel.equals(channel) && this.client.equals(client)` is cleaner. There is no requirement to continually check if the method parameters are non-null, because it does not matter.

Reply:
Fixed